src/bindgen/cdecl.rs

```diff
@@ -213,14 +215,18 @@ impl CDecl {
     }
 }
 
-pub fn write_func<F: Write>(out: &mut SourceWriter<F>, f: &Function, layout_vertical: bool) {
-    &CDecl::from_func(f, layout_vertical).write(out, Some(&f.name));
+pub fn write_func<F: Write>(out: &mut SourceWriter<F>, f: &Function, layout_vertical: bool, documentation: bool) {
+    if documentation {
+        &CDecl::from_func(f, layout_vertical).write(out, Some(&f.name), &f.doc_comment);
+    } else {
+        &CDecl::from_func(f, layout_vertical).write(out, Some(&f.name), &Documentation::none());
+    }
 }
 
-pub fn write_field<F: Write>(out: &mut SourceWriter<F>, t: &Type, ident: &str) {
-    &CDecl::from_type(t).write(out, Some(ident));
+pub fn write_field<F: Write>(out: &mut SourceWriter<F>, t: &Type, ident: &str, doc: &Documentation) {
```

Comment:
Can we instead have the callers print the documentation before calling `write_field`? I'd like to keep these modules separate.